packages/ember-views/lib/attr_nodes/attr_node.js

```diff
@@ -56,29 +56,45 @@ AttrNode.prototype.render = function render(buffer) {
 
   var value = read(this.attrValue);
 
-  if (this.attrName === 'value' && (value === null || value === undefined)) {
-    value = '';
-  }
-
   if (value === undefined) {
     value = null;
   }
 
+  if (this.attrName === 'value') {
+    if (value === null) {
+      value = '';
+    }
 
-  // If user is typing in a value we don't want to rerender and loose cursor position.
-  if (this.hasRenderedInitially && this.attrName === 'value' && this._morph.element.value === value) {
-    this.lastValue = value;
-    return;
+    if (this.hasRenderedInitially && this._morph.element.value === value) {
+      // If user is typing in a value we don't want to rerender and lose cursor position.
+      this.lastValue = value;
+      return;
+    }
   }
 
   if (this.lastValue !== null || value !== null) {
-    this._deprecateEscapedStyle(value);
-    this._morph.setContent(value);
+    if (this.attrName === "class" && this.lastValue !== this._morph.element.className && this._morph.element.className && typeof this._morph.element.className === "string") {
```

Comment:
I agree. I was concerned about making this as performant as possible because I thought this method is probably called quite a lot, so I set it up to try to filter out as much as I could before adding any new variables. I'll just create the variable.